cli/command/stack/kubernetes/stackclient.go

```diff
@@ -0,0 +1,154 @@
+package kubernetes
+
+import (
+	"fmt"
+
+	composev1beta1 "github.com/docker/cli/kubernetes/client/clientset/typed/compose/v1beta1"
+	composev1beta2 "github.com/docker/cli/kubernetes/client/clientset/typed/compose/v1beta2"
+	"github.com/docker/cli/kubernetes/labels"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+// stackClient talks to a kubernetes compose component.
+type stackClient interface {
+	CreateOrUpdate(s stack) error
+	Delete(name string) error
+	Get(name string) (stack, error)
+	List(opts metav1.ListOptions) ([]stack, error)
+	IsColliding(servicesClient corev1.ServiceInterface, s stack) error
+}
+
+// stackV1Beta1 implements stackClient interface and talks to compose component v1beta1.
+type stackV1Beta1 struct {
+	stacks composev1beta1.StackInterface
+}
+
+func (c *KubeCli) newStackV1Beta1() (stackClient, error) {
```

Comment:
I would rather have `func newStackV1Beta1(config *rest.Config, namespace string) …` here (not tying it up with `KubeCli`)

Reply:
ok, fixed